File: ZayirAlkhayr.Entities/Reports/SearchReportModel.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ZayirAlkhayr.Entities.Common;

namespace ZayirAlkhayr.Entities.Reports
{
    public class SearchReportModel
    {
        public string ReportType { get; set; }
        public bool IsLandScape { get; set; }
        public int RowCount { get; set; }
        public string UserName { get; set; }
        public List<QueryString> QueryString { get; set; } = new();
        public List<FilterModel> FilterItems { get; set; } = new();
        public List<PDFHeaderSelected> Headers { get; set; } = new();
    }

    public class QueryString
    {
        public string Key { get; set; }
        public string Value { get; set; }
    }
}

File: ZayirAlkhayr.Angular/src/app/Models/shared/SearchReportModel.ts
import { StudentPrintSlot } from "../school/student/StudentPrintSlot";
import { FilterModel } from "./FilterModel";

export interface SearchReportModel {
    reportType: string;
    isLandScape?: boolean;
    rowCount?: number;
    userName?: string;
    queryString?: QueryString[];
    filterItems?: FilterModel[];
    headers?: PDFHeaderSelected[];
    studentCards?:StudentPrintSlot[];
}

export interface QueryString {
    key: string;
    value: string;
}

export interface PDFHeaderSelected {
    nameEn: string;
    nameAr: string;
    isAllowSummation: boolean;
    valueType: string;
    displayOrder: number;
    isSelected: boolean;
}